pkg/functions/function_structure.go

```diff
@@ -0,0 +1,22 @@
+package functions
+
+import "encoding/json"
+
+type Item struct {
+	Type       string                 `json:"type"`
+	Properties map[string]interface{} `json:"properties"`
+}
+
+type JSONFunctionStructure struct {
+	OneOf []Item                 `json:"oneOf,omitempty"`
+	AnyOf []Item                 `json:"anyOf,omitempty"`
+	Defs  map[string]interface{} `json:"$defs,omitempty"`
+}
+
+func (j JSONFunctionStructure) Grammar(options ...func(*GrammarOption)) string {
+	grammarOpts := &GrammarOption{}
+	grammarOpts.Apply(options...)
+
+	dat, _ := json.Marshal(j)
```

Comment:
Is it a good idea to just discard errors here? These functions eventually operate on unconstrained user input... so I'd like to be robust against any cycles or worse.